### PR TITLE
Disallow definition objects in configs, except ReferenceInterface

### PR DESCRIPTION
**Pull request by yiiliveext**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | #83

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,14 +17,14 @@
     "minimum-stability": "dev",
     "require": {
         "php": "^7.4|^8.0",
-        "psr/container": "1.0.0",
-        "yiisoft/injector": "^1.0"
```
Comment (samdark) [inline composer.json:21]: It is used in CallableDefinition. Can't be moved to dev.

File: src/Definition/DefinitionResolver.php [hunk 2]
```diff
@@ -65,11 +66,12 @@ public static function resolve(ContainerInterface $container, $definition)
...
     {
-        if ($value instanceof DefinitionInterface || is_array($value)) {
+        if ($value instanceof ReferenceInterface || is_array($value)) {
             return $value;
         }
-        if (!is_string($value) && is_callable($value, true)) {
-            return new CallableDefinition($value);
+
+        if ($value instanceof DefinitionInterface) {
+            throw new InvalidConfigException('Only reference allowed in parameters, the definition object received:' . var_export($value, true));
```
Comment (samdark) [inline src/Definition/DefinitionResolver.php:74]:
```suggestion
            throw new InvalidConfigException('Only references are allowed in parameters, a definition object was provided: ' . var_export($value, true));
```

**Review by samdark — CHANGES_REQUESTED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
More configs should be fixed:

- https://github.com/yiisoft/translator-message-php/blob/master/config/common.php
- https://github.com/yiisoft/translator-extractor/blob/master/config/console.php#L14
- https://github.com/yiisoft/translator-message-gettext/blob/master/config/common.php#L13
- https://github.com/yiisoft/yii-db-migration/blob/master/config/console.php#L29
-

File: src/Definition/DefinitionResolver.php [hunk 3]
```diff
@@ -65,11 +66,12 @@ public static function resolve(ContainerInterface $container, $definition)
...
     public static function ensureResolvable($value)
     {
-        if ($value instanceof DefinitionInterface || is_array($value)) {
+        if ($value instanceof ReferenceInterface || is_array($value)) {
             return $value;
         }
-        if (!is_string($value) && is_callable($value, true)) {
-            return new CallableDefinition($value);
+
+        if ($value instanceof DefinitionInterface) {
```
Comment (vjik) [inline src/Definition/DefinitionResolver.php:73]: I think this checking is useless... On ArrayDefinition throws exception without this check.

Comment (yiiliveext) [reply src/Definition/DefinitionResolver.php:73]: No it isn't useless.

**Pull request merged**